docs/McpClient-Feedback.md

```diff
@@ -0,0 +1,47 @@
+ModelContextProtocol.Protocol.Implementation name is too generic.  Lots of things in this namespace named poorly.  That might be OK if users don't have to deal with them, but it looks like they do.
+
+
+ModelClient.CompleteAsync - naming seems off.  Maybe "GetCompletion" 
+
+
+
+static ModelClient.CreateSamplingHandler - why is this factory method on ModelClient???  Maybe should be an extension method.
+
+
+
+ModelClient - Enumerate\* vs List\* - why have these redundant methods that return the same thing just one wrapping.  Can we just return IAE and have folks materialize that?
+
+
+
+ModelClient.EnumerateToolsAsync - why accept JsonSerializerOptions on this call, it doesn't feel like something tied to enumerating tools.  Would it make more sense to make this a property on the client?
```

Comment:
It gets stored into the create McpClientTool instance (McpClientTool : AIFunction), where it's needed for serializing arguments/results when the AIFunction is invoked.